### ISSUE TITLE
Edit and apply the config map

### ISSUE THREAD

**Issue by dinar-dalvi** [kubevious/kubevious#7]
Please add the ability to edit and update the config map

**Comment by rubenhak** [comment]
@dinar-dalvi, I'd like to collect other requests in regards with configuration and making changes to the cluster before jumping in to solution. Currently Kubevious only requests read-only access to the cluster. Requesting write permissions at this stage might significantly affect adoption rate.

Please tell more about your use case. Do you want to modify any config map? Are those mapped as volumes or used as environment variables? If there are other environment variables set on pod specs, would you like to modify those as well? 

Based on your and others' use-cases the solution might be a extension would let ones to edit the cluster, or download a patch file to be applied through kubectl, or something else.

**Comment by dinar-dalvi** [comment]
My use case as follows, we have a need to modify the config-map on the fly for some services. Post modify I can sale the pods down and back up so the change is picked up. For now i can do that doing kubectl but want to see if it's possible from the ui.

**Comment by rubenhak** [comment]
ok, got that. Transition from viewer to editor as major change. I'll keep the ticket open for some time to collect other feedback. Based on other use cases will be able to generalize. Thanks

**Comment by obeyler** [comment]
If you give the opportunity to edit a config map you also need to give the opportunity to kill a pod as the config map is only loaded during start

**Comment by rubenhak** [comment]
@obeyler, good poing. We have a different solution for that. We are going to show the pods/deployments that are "out-of-sync" from configmaps. That should work regardless how the configmap was modified. But that is different from this feature request

**Comment by rubenhak** [comment]
@dinar-dalvi, @obeyler, while Kubevious is still using a read-only access to Kubernetes, we added capability to sanitize the YAML files (remove generation, uid, selflink, status, etc files), edit the file in place, and download edited file ready to be applied with kubectl or to be checked in to git (of you use something similar to GitOps). Kubevious also creates the kubectl command to be copy-pasted to terminal for quick execution.

Here is a quick screen recording of the capability: https://cln.sh/Ecr1u6

```
helm upgrade --atomic -i -n kubevious \
    --version 0.7.4 \
    --set ingress.enabled=true \
    kubevious kubevious/kubevious
```

**Comment by obeyler** [comment]
[2020-08-07 15:55:01.635 +0000] INFO  (LocalLoader/1 on kubevious-parser-696d46c7bd-88489): [setupApiGroup] Lease :: coordination.k8s.io :: v1 :: leases...
***** ERROR *****
{
  status: 503,
  message: 'Service Unavailable',
  method: 'GET',
  baseURL: 'https://10.245.0.1:443',
  url: '/apis/metrics.k8s.io/v1beta1'
}
[2020-08-07 15:55:01.640 +0000] ERROR (Context/1 on kubevious-parser-696d46c7bd-88489): 
    status: 503
    message: "Service Unavailable"
    method: "GET"
    baseURL: "https://10.245.0.1:443"
    url: "/apis/metrics.k8s.io/v1beta1"
So I think the pb is not on your part it may due to my last coredns version fix ;-)
Thanks ruben for the log who help me understand

**Comment by rubenhak** [comment]
@obeyler, thanks for trying out. Looks like your response was related to https://github.com/kubevious/parser/issues/4

Did you have a chance to take a look to editing capability? Does it help with making changes to configs?

**Comment by pedroparraortega** [comment]
i do not think kubevious (cluster one) should manage resources.... At least as it is right now. In our case, ie, we have a multitenant cluster where we have users who can use kubevious to check their workloads together with cluster admins. Allowing the UI to do changes on the fly would require a security layer in top probably based on rbacs, etc.

**Comment by rubenhak** [comment]
> i do not think kubevious (cluster one) should manage resources.... At least as it is right now. In our case, ie, we have a multitenant cluster where we have users who can use kubevious to check their workloads together with cluster admins. Allowing the UI to do changes on the fly would require a security layer in top probably based on rbacs, etc.

Letting Kubevious to modify resources would change the use model and would potentially conflict with existing processing like GitOps, etc. 

@pedroparraortega, did you have a change to take a look to convenience capability to allow to modify the config in UI for download and copy-paste purposes. UI also sanitizes the config removing unnecessary props like uid, generation, selflink, status, etc.

**Comment by pedroparraortega** [comment]
> > i do not think kubevious (cluster one) should manage resources.... At least as it is right now. In our case, ie, we have a multitenant cluster where we have users who can use kubevious to check their workloads together with cluster admins. Allowing the UI to do changes on the fly would require a security layer in top probably based on rbacs, etc.
> 
> Letting Kubevious to modify resources would change the use model and would potentially conflict with existing processing like GitOps, etc.
> 
> @pedroparraortega, did you have a change to take a look to convenience capability to allow to modify the config in UI for download and copy-paste purposes. UI also sanitizes the config removing unnecessary props like uid, generation, selflink, status, etc.

i do not see any issues by allowing the UI do modifications always that those are not applied to the cluster.

@obeyler maybe for what you want to do, you can check this project [Wave](https://github.com/pusher/wave) which allow you to configure dependent resources and the this will trigger the auto update of those objects

**Comment by kubevious** [comment]
Resolved in [v0.7](https://github.com/kubevious/kubevious/releases/tag/v0.7).
